packages/components/src/Dialog/Layout/DialogHeader.tsx

```diff
@@ -24,102 +24,46 @@
 
  */
 
-import { useTranslation } from 'react-i18next'
-import React, { FC, ReactNode, useContext } from 'react'
+import React, { FC, useContext } from 'react'
 import styled from 'styled-components'
-import {
-  CompatibleHTMLProps,
-  omitStyledProps,
-  SpaceProps,
-  space,
-  reset,
-  FontSizeProps,
-  FontWeightProps,
-} from '@looker/design-tokens'
-import { Close } from '@styled-icons/material/Close'
-import { IconButton } from '../../Button'
-import { Heading } from '../../Text'
 import { DialogContext } from '../DialogContext'
+import { ModalHeader, ModalHeaderProps } from '../../Modal/ModalHeader'
+import { CloseHeaderButton } from '../../Modal/CloseHeaderButton'
 
-export interface DialogHeaderProps
-  extends SpaceProps,
-    CompatibleHTMLProps<HTMLElement>,
-    FontSizeProps,
-    FontWeightProps {
-  children: ReactNode
+type WithDetail = {
+  detail?: ModalHeaderProps['detail']
+  hideClose?: never
+}
+
+type WithHideClose = {
+  detail?: never
   /**
    * Don't include the "Close" option
    * @default false
    */
   hideClose?: boolean
-  /**
-   * Replaces the built-in `IconButton` (generally used for close) with an arbitrary ReactNode
-   */
-  detail?: ReactNode
 }
 
+export type DialogHeaderProps = WithDetail | WithHideClose
```

Comment:
This needs to extended the type for `ModalHeader`:

```suggestion
export type DetailUnion = WithDetail | WithHideClose

export type DialogHeaderProps = DetailUnion & Omit<ModalHeaderProps, 'detail'>
```